### PR TITLE
chore: Migrated project to AndroidX

### LINKED ISSUE #2860: convert the app from v7 support versions to androidx
v7 support version is going to be deprecated soon and we have to migrate the app to androidx

### PR DESCRIPTION
**Pull request by prudhvir3ddy**
Fixed #2860 

Changes: [ replaced all the support versions to androidx versions ]

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,4 +12,4 @@ build
 /app/src/main/jni/include
 libopencv_*
 viewflow/
-keystore.properties
+keystore.properties
```
Comment (abishekvashok) [inline .gitignore]: Please revert this change, a new line at eof never hurts :P

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,5 +1,6 @@
+#Wed Sep 11 10:09:54 IST 2019
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.1.1-all.zip
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.4.1-all.zip
```
Comment (codedsun) [inline gradle/wrapper/gradle-wrapper.properties]: Are we also upgrading the gradle plugin version?

Comment (yashk2000) [reply gradle/wrapper/gradle-wrapper.properties]: We should update the `complieSdkVersion`, `targetSdkVersion` and the `buildToolsVersion` too. This is a different issue though. 
A separate issue and pr should be created for this.

Comment (codedsun) [reply gradle/wrapper/gradle-wrapper.properties]: @yashk2000 Shall I take this?

**Comment by yashk2000** [comment]
@abishekvashok @iamareebjamal

File: gradle.properties [hunk 3]
```diff
@@ -25,3 +25,6 @@ org.gradle.parallel=true
 # The developer environment is optimized for speed and feedback so we nearly always run Gradle jobs with the daemon.
 android.enableBuildCache=true
 org.gradle.daemon=true
+android.useAndroidX=true
+android.enableJetifier=true
+android.jetifier.blacklist = butterknife-compiler
```
Comment (iamareebjamal) [inline gradle.properties]: Use latest butterknife

File: build.gradle [hunk 4]
```diff
@@ -44,8 +46,8 @@ allprojects {
 }
 
 ext {
-    supportLibraryVersion = '28.0.0'
-    butterknifeVersion = '8.6.0'
+    supportLibraryVersion = '1.1.0'
+    butterknifeVersion = '9.0.0-SNAPSHOT'
```
Comment (iamareebjamal) [inline build.gradle]: Why using snapshot? Latest butterknife supports AndroidX

File: gradle.properties (hunk 3 above)
Comment (iamareebjamal) [reply gradle.properties]: Also, add a newline at the last line

File: build.gradle (hunk 4 above)
Comment (prudhvir3ddy) [reply build.gradle]: sorry i missed this. updated thanks.

File: gradle.properties (hunk 3 above)
Comment (prudhvir3ddy) [reply gradle.properties]: done. by the way, what is the love for adding a line?

Comment (iamareebjamal) [reply gradle.properties]: https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline
